### ISSUE TITLE
Calling MakeCurrent() when no context is created crashes the window.

### ISSUE THREAD

**Issue by emmauss** [ForeverZer0/glfw-net#3]
When creating a gamewindow for vulkan, and one decides to disable contexts using `Glfw.WindowHint(Hint.ClientApi, ClientApi.None);`, the app crashes with a `The specified window has no contex` error at `MakeCurrent()` in the constructor. Currently, the only way to use vulkan is if you use the raw functions to create and manage the window.

**Comment by jpbruyere** [comment]
MakeCurrent is for multithreading with opengl **only**. Multithreading in vulkan is done with proper commands and descriptors allocations.

**Comment by emmauss** [comment]
The thing is, GameWindow would always call MakeCurrent in it's constructor. and if you make a contexted gl window, vulkan can't render to it as it is owned by ogl.

**Comment by ForeverZer0** [comment]
As stated above, `MakeCurrent` is only for OpenGL or OpenGL ES context. The `GameWindow` class calls does indeed invoke `MakeCurrent` internally within the constructor, which is an oversight of mine. I will submit a fix for this very soon. 

It is still possible using the static `Glfw` class directly, though you would lose the obvious benefit of the more intuitive `GameWindow` class.

**Comment by ForeverZer0** [comment]
Added a check in the constructor before calling `MakeCurrent` that the window context was either OpenGL or OpenGL ES (i.e. NOT Vulkan). I did not add a check to `SwapBuffers`, as this is the responsibility of the application to not call when using Vulkan, and is not used automatically within the `GameWindow` class. 

[Context-less Windows](https://www.glfw.org/docs/latest/context_guide.html#context_less)
